website/project/model.py

```diff
@@ -162,6 +162,31 @@ class Comment(GuidStoredObject):
     # }
     reports = fields.DictionaryField(validate=validate_comment_reports)
 
+    # For Django compatibility
+    @property
+    def pk(self):
+        return self._id
+
+    @property
+    def absolute_api_v2_url(self):
+        return absolute_reverse('comments:comment-detail', kwargs={'comment_id': self._id})
+
+    # used by django and DRF
+    def get_absolute_url(self):
+        return self.absolute_api_v2_url
+
+    def return_content(self, auth):
+        """ Returns the comment content if the user is allowed to see it. Deleted comments
+        can only be viewed by the user who created the comment."""
+        if (not auth or auth.user.is_anonymous()) and not self.node.is_public:
+            raise PermissionsError
+
+        if self.is_deleted and (((not auth or auth.user.is_anonymous()) and self.node.is_public)
+                                or (auth and not auth.user.is_anonymous() and self.user._id != auth.user._id)):
+            return 'Comment deleted.'
```

Comment:
The text that is shown for deleted comments is a UI/front-end concern and therefore doesn't belong on the model. Consider that we might want to stylize the deleted text in a special way. Also, what if a comment actually contains the text "Comment deleted."? 

Better to return a `None` here, and let the client display deleted comments as they see fit.